lib/blockscore/actions/retrieve.rb

```diff
@@ -16,7 +16,7 @@ module ClassMethods
         def retrieve(id, options = {})
           fail ArgumentError, 'ID must be supplied' if id.nil? || id.empty?
           fail ArgumentError, 'ID is malformed' unless id =~ RESOURCE_ID_FORMAT
-          req = ->() { get("#{endpoint}/#{id}", options) }
+          req = ->() { get(endpoint + id, options) }
           new(id: id, &req)
```

Comment:
In a future refactoring I would suggest moving the line above into this line so it becomes:

```ruby
new(id: id) { get(endpoint + id, options) }
```

There doesn't seem to be any point in defining a separate lambda.

Also, you could yield the object into the block without breaking the public API (because anyone using this api now would not be yielding anything to the block, which would not raise an error). This would allow you to use `#member_endpoint`, since the `id` would've been set in the constructor. So this could then become:

```ruby
new(id: id) { |object| get(object.member_endpoint, options) }
```